src/main/java/persistence/sql/common/meta/TableName.java

```diff
@@ -1,39 +1,48 @@
 package persistence.sql.common.meta;
 
+import jakarta.persistence.Entity;
 import jakarta.persistence.Table;
+import persistence.exception.InvalidEntityException;
 
-import java.lang.annotation.Annotation;
+public class TableName {
 
-class TableName {
     private final String value;
 
     private <T> TableName(Class<T> tClass) {
         this.value = parseTableName(tClass);
     }
 
-    protected static <T> TableName of(Class<T> tClass) {
+    public static <T> TableName of(Class<T> tClass) {
         return new TableName(tClass);
     }
 
     /**
      * class의 이름을 가져와 table 이름으로 설정합니다.
      */
-    <T> String parseTableName(Class<T> tClass) {
-        Class<Table> annotation = Table.class;
-        String tableName = tClass.getSimpleName();
-
-        if (isTable(tClass, annotation)
-                && !"".equals(tClass.getAnnotation(annotation).name())) {
-            tableName = tClass.getAnnotation(annotation).name();
+    private <T> String parseTableName(Class<T> tClass) {
+        if (isNotEntity(tClass)) {
+            throw new InvalidEntityException();
         }
 
-        return tableName;
+        return getTableName(tClass);
     }
 
-    private static <T, A> boolean isTable(Class<T> tClass, Class<A> aClass) {
-        Class<? extends Annotation> annotation = aClass.asSubclass(Annotation.class);
+    private <T> boolean isTable(Class<T> tClass) {
+        return tClass.isAnnotationPresent(Table.class);
+    }
+
+    private <T> boolean isNotEntity(Class<T> tClass) {
+        return !tClass.isAnnotationPresent(Entity.class);
+    }
 
-        return tClass.isAnnotationPresent(annotation);
+    private <T> String getTableName(Class<T> tClass) {
+        String tableName = tClass.getSimpleName();
+
+        if (isTable(tClass) && !"".equals(tClass.getAnnotation(Table.class).name())) {
```

Comment:
```suggestion
        if (isValidTableName(tClass)) {
```
이미 isTable 이라는 메소드로 분리 해주셨는데 
조건을 묶어주는건 어떻게 생각 하시나요?